### PR TITLE
chore: move setup requirements from setup.py to pyproject.toml

### LINKED ISSUE #5346: chore: DEPRECATION WARNING when installing Jina
```shell
DEPRECATION: jina is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
```

### PR DESCRIPTION
**Pull request by Jackmin801**
**Goals:**

- resolves #5346

### CODE REVIEW

**Comment by Jackmin801** [comment]
I don't actually know if it works because I can't reproduce the deprecation warning. What I can see is that the first few lines in pip install have changed

Old:
image

New:
image

**Comment by Jackmin801** [comment]
The first commit -- [fix: move the easy stuff to setup.cfg](https://github.com/jina-ai/jina/pull/5351/commits/e9844d3ddb0d2f424129b9ac0e8ed68fb24354d2) might also be unnecessary

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,49 @@
+[metadata]
+name = jina
+version = 3.11.1
```
Comment (JoanFM) [inline setup.cfg:3]: we need to adapt our CI/CD to update this `version` then

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 1 above)
Comment (Jackmin801) [reply setup.cfg:3]: yup

**Comment by Jackmin801** [comment]
Hrmm the change broke some weird things. Let me try a smaller change. I think this might be enough to resolve the warning already. The first few lines of the pip install already use pyproject.toml

File: setup.cfg (hunk 1 above)
Comment (Jackmin801) [reply setup.cfg]: we can do this later. I think the current change of just moving the build configs is enough

**Comment by Jackmin801** [comment]
Can confirm this fix works. Inside a python environment without `wheel` installed, the deprecation warning appears without the commit in this PR. With the commit in this PR, the deprecation warning does not appear.

**Review by JoanFM — APPROVED** [review]

**Pull request merged**